site/source/tutorials/005_form_widgets/index.md

```diff
@@ -88,35 +89,39 @@ Let's use those classes and a few virtual DOM nodes, to add the visual elements
 	}
 ```
 
-At this point, the user interface for the form is available, but it doesn't do anything since we haven't specified any event handlers. In the [last](../004_user_interactions/) tutorial, we learned how to add event handlers to custom widgets by assigning a method to an event. When using pre-built widgets, however, we pass the handlers as properties. For example, we are going to need a way to handle each text fields `change` event. To do that, we provide the desired handler function as the `onChange` property that is passed to the widget. Update the `render` method once again:
+At this point, the user interface for the form is available, but it does not do anything since we have not specified any event handlers. In the [last tutorial](../004_user_interactions/), we learned how to add event handlers to custom widgets by assigning a method to an event. When using pre-built widgets, however, we pass the handlers as properties. For example, we are going to need a way to handle each text field's `change` event. To do that, we provide the desired handler function as the `onChange` property that is passed to the widget. Update the `render` method once again:
 
 {% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:33-110 %}
 
-This form of the `render` method now does everything that we need it to: it creates the user interface and registers the event handlers that will update the application as the user enters information.
+This form of the `render` method now does everything that we need from: it creates the user interface and registers the event handlers that will update the application as the user enters information.
 
-The method starts by decomposing the properties down in to local constants. We still need to define those properties, so update the `WorkerFormProperties` interface to include them:
+The `render` method starts by decomposing the properties into local constants. We still need to define those properties, so update the `WorkerFormProperties` interface to include them:
 
 {% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:9-19 %}
 
 There are three types of properties that we are using in this form. The `firstName`, `lastName` and `email` properties are going to set the values that are displayed in the form fields. The `firstNameInvalid`, `lastNameInvalid` and `emailInvalid` properties will be used to indicate if a field contains invalid data (e.g. an invalid email address in the `email` field). Finally, the `onChange`, `onBlur` and `onSubmit` properties expose the events that the `WorkerForm` widget can emit. To see how these different property types are used, let's examine the properties that are being passed into first `TextInput` widget:
```

Comment:
"let's examine the properties that are being passed into first `TextInput` widget:" -> "let's examine the properties that are passed into first `TextInput` widget:"